src/Components/Calculator.js

```diff
@@ -1,121 +1,89 @@
-import React from 'react';
+import React, { useState } from 'react';
 import calculate from '../logic/calculate';
 
-class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      element: {
-        total: null,
-        next: null,
-        operation: null,
-      },
-    };
-    this.result = this.result.bind(this);
-  }
+const Calculator = () => {
+  const [element, AddNewEl] = useState({
+    total: null,
+    next: null,
+    operation: null,
+  });
 
-  result(event) {
-    const { element } = this.state;
+  const Calc = (event) => {
```

Comment:
[OPTIONAL] I advise naming this function "calc" rather than "Calc" because the JavaScript method is written in camelCase.